src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -457,6 +460,49 @@ private static string GetValidatedAbsolutePath(string originalPath, out string v
         }
 
         /// <summary>
+        /// Returns a value indicating whether the requested image has been updated.
+        /// </summary>
+        /// <param name="creationDate">The creation date.</param>
+        /// <returns>The <see cref="bool"/></returns>
+        private async Task<bool> IsUpdatedAsync(DateTime creationDate)
+        {
+            bool isUpdated = false;
+
+            try
+            {
+                if (new Uri(this.RequestPath).IsFile)
+                {
+                    FileInfo imageFileInfo = new FileInfo(this.RequestPath);
+                    if (imageFileInfo.Exists)
+                    {
+                        // Pull the latest info.
+                        imageFileInfo.Refresh();
+
+                        // If it's newer than the cached file then it must be an update.
+                        isUpdated = imageFileInfo.LastWriteTimeUtc > creationDate;
+                    }
+                }
+                else
+                {
+                    // Try and get the headers for the file, this should allow cache busting for remote files.
+                    HttpWebRequest request = (HttpWebRequest)WebRequest.Create(this.RequestPath);
+                    request.Method = "HEAD";
+
+                    using (HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync())
+                    {
+                        isUpdated = response.LastModified.ToUniversalTime() > creationDate;
+                    }
+                }
+            }
+            catch
+            {
```

Comment:
There's no way for me to determine whether the request is from a remote url or a db or whatever IImageService implementation someone has created that isn't local. This was the cheapest way I could think of. Logging would probably just bloat the log when a custom service is used.